src/game/combat.h

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright � 2020 uwadmin12
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include <unordered_set>
+#include <map>
+
+#include "types.h"
+#include "object/creature.h"
+
+namespace reone {
+
+namespace game {
+
+class Area;
+
+class CombatManager {
```

Comment:
Just `Combat` would be a better name, I believe. Manager classes are considered an anti-pattern, as it is too vague of a term.